modules/adapter-bookshelf/index.js

```diff
@@ -249,8 +257,12 @@ Adapter.prototype.read = function (opts, mode) {
       columns: fields,
       withRelated: _.intersection(self.relations(), opts.include || [])
     }).then(function (result) {
+      result.mode = mode;
```

Comment:
Could we leave a note here saying that we want to formalize all the stuff represented here into some "thing"?